Add more background. (#7)
jhoyla/draft-jhoyla-tls-extended-key-schedule: draft-jhoyla-tls-extended-key-schedule.md

permit the use of combined shared secrets, possibly from multiple key exchange
algorithms, to be included in the key schedule. This pattern is common for Post
Quantum key exchange algorithms, as discussed in

> {{?I-D.ietf-tls-hybrid-design}}. {{?I-D.ietf-tls-hybrid-design}} uses the same
> pattern as described in this draft, but does not add the requisite framing.

Reviewer: ```suggestion
{{?I-D.ietf-tls-hybrid-design}}. In particular, {{?I-D.ietf-tls-hybrid-design}} uses the
concatenation pattern described in this draft, but does not add the requisite framing.
```

Making {{?I-D.ietf-tls-hybrid-design}} compliant with this design merely
requires adding the 6-byte framing prefix to the key. Because the secret input
is never sent on the wire neither side needs to parse it, it is constructed

> directly by both the client and server.

Reviewer: ```suggestion
```

--- back

> # Sample Potential Use Cases

Reviewer: ```suggestion
# Potential Use Cases
```

> This structure can be used to implement a number of different effects naturally.

Reviewer: ```suggestion
```

> The draft provides an importer interface that mirrors the exporter interface.
> This provides a way to layer TLS on to other protocols, as opposed to layering
> other protocols over TLS. This can be used for hybrid key exchanges, which, in
> effect, is layering TLS over a secondary AKE. Although the key exchanges are
> interleaved, the post-quantum AKE completes first, as demonstrated by its output
> key being used as an input for computing TLS's master secret.

Reviewer: This is a bit misleading. The exporter interface is, well, an actual interface. This document does not provide an *explicit* importer interface, but a way for other drafts to add those interfaces. I'd suggest:

```suggestion
The draft provides a mechanism for importing additional information into the TLS key schedule.
Future applications and specifications can use this mechanism to layer TLS on to other protocols, 
as opposed to layering other protocols over TLS. For example, as discussed in {{intro}}, this can 
be used for hybrid key exchange, which, in effect, is layering TLS over a secondary AKE. 
Although the key exchanges are interleaved, the post-quantum AKE completes first, as demonstrated 
by its output key being used as an input for computing TLS's master secret.
```

This can also be used in more direct ways, such as bootstrapping EAP-TLS as in
{{?I-D.friel-tls-eap-dpp}}. This draft also allows for more direct

> implementations of things such as semi-static DH. The aim of this draft is to be

Reviewer: ```suggestion
implementations of things such as semi-static DH {{?I-D.ietf-tls-semistatic-dh}}. The aim of this draft is to be
```
